numpy/lib/_iotools.py

```diff
@@ -50,6 +54,13 @@ def _is_bytes_like(obj):
     return True
 
 
+def _is_path(obj):
+    """
+    Check whether obj is a pathlib.Path object.
+    """
+    return PurePath is not None and isinstance(obj, PurePath)
```

Comment:
`PurePath` looks like the wrong type to check as it may be purely abstract, `Path` would seem better.